### DISCUSSION TITLE
How to properly exit fragment with sceneview

### DISCUSSION THREAD

**Discussion by HajdukG** [sceneview/sceneview#463] [Q&A]
How to properly exit fragment? If i navigate out - the app simply crashes. What steps should i take to make sure i can enter and exit fragment safely?

**Comment by andytriboletti** [comment]
I also encountered this problem.

**Reply by andytriboletti** [reply]
See my comment on this thread:
https://github.com/SceneView/sceneview-android/issues/450

**Reply by mrkjrcxcnd** [reply]
tysm. I finally fixed it.

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, if you're using SceneView in a Compose-based app (recommended), resource cleanup is handled automatically when the composable leaves the composition.

If you're using it in a Fragment with `ComposeView`, Compose lifecycle handles disposal:
```kotlin
class MyFragment : Fragment() {
    override fun onCreateView(...) = ComposeView(requireContext()).apply {
        setContent {
            SceneView { /* nodes */ }
        }
    }
    // No manual cleanup needed — Compose handles it
}
```

If you experience crashes on fragment exit, make sure you're on v4.0.0-rc.1 which includes lifecycle fixes, particularly for ViewNode and resource disposal.
